galasa-simplatform-application/src/main/java/dev/galasa/simplatform/listener/ListenerManager.java

```diff
@@ -0,0 +1,163 @@
+ * 
+ */
+
+public class ListenerManager implements IListener{
+	
+	private Socket socket;
+	private List<String> headers = new ArrayList<>();
+	private String payload = new String();
+	
+    private Logger log = Logger.getLogger("Simplatform");
+
+	public void run() {
+		try {
+			
+			/* Attempt to read the request listened by the server */
+			processInput();
+			
+			/* Interpret which sort of sub-listener we will need to call upon */
+            String path = findPath().trim();
+            
+            /* Switching through the different path options */
+            switch(path) {
+            
+            case "/updateAccount":   	
+            	/* Initialise the CreditAccountListener and send payload */
+            	new CreditAccountListener().sendRequest(this, payload);
+            	break;
+            	
+            case "/processTransfer":
+            	/* Initialise the AccountTransferListener and send payload */
+            	new AccountTransferListener().sendRequest(this, payload);
+            	break;
+            	
+            default:
+            	/* Request has not been recognised */
+                log.log(Level.WARNING, () -> String.format("Request was not sent to path /updateAccount or /processTransfer, it was: %1$s returning 404", path));
+            	return;
+            }
+            
+		} catch (Exception e) {
+			log.severe("The request was not readable. Something has gone, very... wrong.");
+			e.printStackTrace();
+			return;
+		}
+	}
+	
+	/* Returns the path located in the header of the XML Request */
+    private String findPath() {
+        StringTokenizer st = new StringTokenizer(headers.get(0), " ");
+        if (st.countTokens() == 3) {
+            st.nextToken();
+            return st.nextToken();
+        }
+        return "";
+    }
+    
+    /* Prepares and sends the response passed back from the sub listener */
+    public void sendResponse(String xmlText) throws Exception {
+		OutputStream output;
+		output = socket.getOutputStream();
+		PrintStream ps = new PrintStream(output);
+		ps.println("HTTP/1.1 200 OK");
+		ps.println("Server: Simulplatform 0.3.0");
+		ps.println("Content-Length: " + xmlText.length());
+		ps.println("Content-Type: application/soap+xml; charset=\"utf-8\"\r\n");
+		ps.println(xmlText);
+		ps.println("\r\n");
+		ps.flush();
+		socket.close();
+    }
+	
+    /* Reads the key information from the XML Request */
+	private void processInput() {
+		BufferedReader br = null;
+		try {
+        	log.log(Level.WARNING, () -> String.format("Received HTTP request from address: %1$s", socket.getInetAddress().toString()));
+			InputStream input = socket.getInputStream();
+			br = new BufferedReader(new InputStreamReader(input));
+		} catch (IOException e) {
+			log.warning("Unable to access input stream from HTTP");
+			return;
+		}
+
+		boolean readAllHeaders = false;
+		try {
+			while (!br.ready()) {
+				try {
+					Thread.sleep(1000);
+				} catch (InterruptedException e) {
+				}
```

Comment:
If we catch InterruptedException we need to handle it and not ignore it silently - see https://stackoverflow.com/questions/3976344/handling-interruptedexception-in-java for more details